### DISCUSSION TITLE
A way to automate time picker for android

### DISCUSSION THREAD

**Discussion by Kristina006** [appium/appium#17198] [Q&A]
I need to automate the function where it will select the time(hour and minutes) either by parameters or randomly. Another case to consider is to test invalid times. The elements only have an accessibility id which goes for each hour.

What is the best way to automate this function and reduce code redundancy?

I really appreciate any help you can provide.

![Screen Shot 2022-07-07 at 10 43 55 AM](https://user-images.githubusercontent.com/94417825/178843015-324e2ec0-b4d3-4968-ab9a-6151ee74dbd8.png)

**Comment by jlipps** [comment]
are you able to find and tap on these number elements in an appium script (not the inspector)?

**Reply by Kristina006** [reply]
Yes, you can tap on these numbers and the picker will change accordingly. But I need to write a method that should do it dynamically, for example, with parameters.

**Reply by jlipps** [reply]
I'm not sure what you mean by "dynamically". If the accessibility IDs are standard, then you should be able to write a method that takes a series of numbers as input and then generates taps on those numbers?

**Comment by KazuCocoa** [comment]
If you could use espresso driver, https://github.com/appium/appium-espresso-driver#mobile-setdate and https://github.com/appium/appium-espresso-driver#mobile-settime may help to set data

**Comment by nextlevelbeard** [comment]
I built this some time ago, see if it helps.
https://pastebin.com/QG8DmrbA

Needs to be updated, moved from sync to async mode and never ran it.
It's possible you'll have to tinker to get it working with it but it was stable and considered quite some edge cases regarding time and date.

**Reply by jlipps** [reply]
Very cool!

**Reply by nextlevelbeard** [reply]
Thanks @jlipps
Since Appium ships webdriverio and allows for script execution with a plugin, do you think helpers like these could be shipped and made available with `mobile:` commands?

Something like `mobile: setTime` and `mobile: setDate` with `time` and `date` parameters.

Everybody keeps reinventing or borrowing the wheel to do this sort of stuff.

**Reply by jlipps** [reply]
this would be a perfect Appium 2 server plugin! call it `appium-android-datetime-plugin` or something. if you can get it mostly put together happy to provide some feedback. looks like you know nodejs already and have the code written in nodejs already so it would mostly be a matter of converting the calls to use appium's internal commands rather than from the outside via webdriverio. and then yeah exposing them as `mobile: ` methods would make sense.

**Reply by kopax-polyconseil** [reply]
We also seek how to test react-native-date-picker and it is surprising difficult to find a way to do so =/
